[10.0][MIG] port procurement_purchase_groupby_analytic to v10.0

Cedric-Pigeon opened the pull request:
Port of the unmerged PR https://github.com/OCA/purchase-workflow/pull/161 to V10.0

The V8.0 PR wasn't merged because of a dependency on an Odoo PR suggesting a hook.
As hooks are accepted with difficulty by Odoo, I changed the way of work to avoid it in V10 by using a `with_context` call.

Commit:
add module to group procurement by analytic

Group purchase order created from procurement by analytic account defined on sale order

Commit:
procurement_purchase_groupby_analytic : set analytic account on group procurement

Commit:
Set analytic account when create purchase order line

Commit:
add setup for procurement_purchase_groupby_analytic

Commit:
[10.0][MIG] port procurement_purchase_groupby_analytic to v10.0

Commit:
[ADD] add setup for procurement_purchase_groupby_analytic

rousseldenis commented on procurement_purchase_groupby_analytic/README.rst:
```diff
@@ -0,0 +1,71 @@
+Current behavior (without this module): add products in existing
+purchase order.
+
+Excepted behavior (with this module): get a separate purchase order for
```
Expected

rousseldenis commented on procurement_purchase_groupby_analytic/__manifest__.py:
```diff
@@ -0,0 +1,35 @@
+    "category": "Other",
+    "website": "http://www.acsone.eu",
+    "depends": ["purchase",
+                "sale_stock",
```
@Cedric-Pigeon As project_id is defined in sale, must appear here.

rousseldenis commented on procurement_purchase_groupby_analytic/__manifest__.py:
```diff
@@ -0,0 +1,35 @@
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Please adapt this according to https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py

rousseldenis commented on procurement_purchase_groupby_analytic/models/procurement.py:
```diff
@@ -0,0 +1,61 @@
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Same as above

rousseldenis commented on procurement_purchase_groupby_analytic/models/procurement.py:
```diff
@@ -0,0 +1,61 @@
+        compute='_compute_analytic_id',
+        comodel_name='account.analytic.account')
+
+    @api.one
```
Deprecated

rousseldenis commented on procurement_purchase_groupby_analytic/tests/test_procurement_groupby_analytic.py:
```diff
@@ -0,0 +1,157 @@
+#
+##############################################################################
+
+import openerp.tests.common as common
```
odoo

rousseldenis commented on procurement_purchase_groupby_analytic/models/purchase.py:
```diff
@@ -0,0 +1,18 @@
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        account_analytic_id = self._context.get('account_analytic_id')
```
self.env.context

rousseldenis reviewed (changes requested):
Code review. Some comments

rousseldenis committed:
[IMP] procurement_purchase_groupby_analytic: Improve behaviour

Change api.one => api.multi, improve search
Avoid changing context in procurement _run and overriding purchase.order search()

rousseldenis committed:
[IMP] procurement_purchase_groupby_analytic: Refactoring
Better OCA guidelines compliance
Use README generator

rousseldenis commented:
@OCA/procurement-purchase-maintainers Needs review

rousseldenis commented:
Travis does not fail because of this PR

rousseldenis commented:
@Cedric-Pigeon I think the code for the computed analytic_account_id on procurement should be avoided by depending on https://github.com/OCA/account-analytic/tree/10.0/sale_procurement_analytic.

Don't you think ? With procurement_analytic, sale_procurement_analytic and purchase_procurement_analytic modules installed, you simplify the code.

Moreover, I think this one should be incompatible with these modules as is.

rousseldenis commented:
@Cedric-Pigeon I think this one is not needed anymore because covered by account_procurement_analytic and particularly this commit : https://github.com/OCA/account-analytic/commit/f6870696416620bd2e8a33d13711fcf44c5eaaf4

rousseldenis committed:
[10.0][FIX] procurement_purchase_groupby_analytic: Change domain if no line

rousseldenis committed:
fixup!

rousseldenis committed:
fixup!
